examples/templates/docker/main.tf

```diff
@@ -2,11 +2,11 @@ terraform {
   required_providers {
     coder = {
       source  = "coder/coder"
-      version = "~> 0.6.12"
+      version = "~> 0.6.17"
     }
     docker = {
       source  = "kreuzwerker/docker"
-      version = "~> 2.22"
+      version = "~> 3.0.1"
```

Comment:
@matifali I checked the docker provider and it fails. Most likely that we need to apply [these migration steps](https://github.com/kreuzwerker/terraform-provider-docker/blob/master/docs/v2_v3_migration.md). For the purpose of this PR, can we bring back the `2.22` version?

```
Error: Missing required argument
The argument "context" is required, but no definition was found.
Error: Unsupported argument
An argument named "path" is not expected here.
```

Reply:
I have followed the migration guide.